### PR TITLE
[Misc] Use registry-based initialization for KV cache transfer connector.

### PR DESCRIPTION
**Pull request by KuntaiDu**
Use registry-based initialization for connector class. This allows people to add third-party connectors without writing a lot of if-else checks.

### CODE REVIEW

File: vllm/config.py [hunk 1]
```diff
@@ -2560,14 +2560,6 @@ def from_cli(cls, cli_value: str) -> "KVTransferConfig":
...
 
     def model_post_init(self, __context: Any) -> None:
-        supported_kv_connector = ["PyNcclConnector", "MooncakeConnector"]
-        if all([
-                self.kv_connector is not None, self.kv_connector
-                not in supported_kv_connector
-        ]):
-            raise ValueError(f"Unsupported kv_connector: {self.kv_connector}. "
-                             f"Supported connectors are "
-                             f"{supported_kv_connector}.")
```
Comment (KuntaiDu) [inline vllm/config.py:2570]: Remove this part of code --- this check will be done inside the kv connector factory class.

**Review by youkaichao — APPROVED** [review]
makes sense, thanks for adding it!

**Comment by youkaichao** [comment]
please also add some tests, examples, docs for it.

I think you can add them after we refactor the plugin tests in https://github.com/vllm-project/vllm/pull/11602

**Comment by KuntaiDu** [comment]
> please also add some tests, examples, docs for it.
> 
> I think you can add them after we refactor the plugin tests in #11602

Get it! Let me take a look at plugin test and see how it works

**Pull request merged**